examples/api/lib/widgets/scroll_position/scroll_controller_on_attach.0.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [ScrollController].
+
+void main() => runApp(const ScrollControllerDemo());
+
+class ScrollControllerDemo extends StatefulWidget {
+  const ScrollControllerDemo({super.key});
+
+  @override
+  State<ScrollControllerDemo> createState() => ScrollControllerDemoState();
+}
+
+class ScrollControllerDemoState extends State<ScrollControllerDemo> {
+  late final ScrollController _controller;
+  bool isScrolling = false;
+
+  void _handleScrollChange() {
+    if (isScrolling != _controller.position.isScrollingNotifier.value) {
+      setState((){
+        isScrolling = _controller.position.isScrollingNotifier.value;
+      });
+    }
+  }
+
+  void _handlePositionAttach(ScrollPosition position) {
+    // From here, add a listener to the given ScrollPosition.
+    // Here the isScrollingNotifier will be used to inform when scrolling starts
+    // and stops and change the AppBar's color in response.
+    position.isScrollingNotifier.addListener(_handleScrollChange);
+  }
+
+  void _handlePositionDetach(ScrollPosition position) {
+    // From here, add a listener to the given ScrollPosition.
+    // Here the isScrollingNotifier will be used to inform when scrolling starts
+    // and stops and change the AppBar's color in response.
+    position.isScrollingNotifier.removeListener(_handleScrollChange);
+  }
+
+  @override
+  void initState() {
+    _controller = ScrollController(
+      onAttach: _handlePositionAttach,
+      onDetach: _handlePositionDetach,
+    );
```

Comment:
attach is called by the Scrollable after the ScrollPosition was created. I'll add more docs.

Reply:
This code sample will be embedded with the docs in scroll_controller.dart. Those docs are meant to provide a lot of the context here rather than duplicating them. If some of this is unclear in the docs on the ScrollController class, we should improve them there rather than in the sample. Let me know if the docs in scroll_controller.dart are not helpful.